pkgs/desktops/gnome-3/core/tracker-miners/default.nix

```diff
@@ -1,47 +1,101 @@
-{ stdenv, intltool, fetchurl, libxml2, upower
-, substituteAll, pkgconfig, gtk3, glib, gexiv2
-, bash, wrapGAppsHook, itstool, vala, sqlite, libxslt
-, gnome3, librsvg, gdk_pixbuf, libnotify
-, evolution-data-server, gst_all_1, poppler
-, icu, taglib, libjpeg, libtiff, giflib, libcue
-, libvorbis, flac, exempi, networkmanager
-, libpng, libexif, libgsf, libuuid, bzip2
-, libsoup, json-glib, libseccomp
-, libiptcdata }:
-
-let
+{ stdenv, fetchurl, substituteAll, intltool, itstool, libxslt, makeWrapper
+, meson, ninja, pkgconfig, vala, wrapGAppsHook, bzip2, dbus, evolution-data-server
+, exempi, flac, giflib, glib, gnome3, gst_all_1, icu, json-glib, libcue, libexif
+, libgsf, libiptcdata, libjpeg, libpng, libseccomp, libsoup, libtiff, libuuid
+, libvorbis, libxml2, poppler, taglib, upower }:
+
+stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
   pname = "tracker-miners";
   version = "2.0.4";
-in stdenv.mkDerivation rec {
-  name = "${pname}-${version}";
 
   src = fetchurl {
     url = "mirror://gnome/sources/${pname}/${gnome3.versionBranch version}/${name}.tar.xz";
     sha256 = "0mp9m2waii583sjgr61m1ni6py6dry11r0rzidgvw1g4cxhn89j6";
   };
 
-  NIX_CFLAGS_COMPILE = "-I${poppler.dev}/include/poppler";
+  # https://github.com/NixOS/nixpkgs/issues/39547
+  LIBRARY_PATH = stdenv.lib.makeLibraryPath [ giflib ];
 
-  enableParallelBuilding = true;
+  nativeBuildInputs = [
+    intltool
+    itstool
+    libxslt
+    makeWrapper
+    meson
+    ninja
+    pkgconfig
+    vala
+    wrapGAppsHook
+  ];
 
-  nativeBuildInputs = [ vala pkgconfig intltool itstool libxslt wrapGAppsHook ];
   # TODO: add libgrss, libenca, libosinfo
   buildInputs = [
-    bzip2 evolution-data-server exempi flac giflib glib gnome3.totem-pl-parser
-    gnome3.tracker gst_all_1.gst-plugins-base gst_all_1.gstreamer icu
-    json-glib libcue libexif libgsf libiptcdata libjpeg libpng libseccomp libsoup
-    libtiff libuuid libvorbis libxml2 poppler taglib upower gexiv2
+    bzip2
+    dbus
+    evolution-data-server
+    exempi
+    flac
+    giflib
+    glib
+    gnome3.gexiv2
+    gnome3.totem-pl-parser
+    gnome3.tracker
+    gst_all_1.gst-plugins-base
+    gst_all_1.gstreamer
+    icu
+    json-glib
+    libcue
+    libexif
+    libgsf
+    libiptcdata
+    libjpeg
+    libpng
+    libseccomp
+    libsoup
+    libtiff
+    libuuid
+    libvorbis
+    libxml2
+    poppler
+    taglib
+    upower
   ];
 
-  LANG = "en_US.UTF-8"; # for running tests
-
-  doCheck = true;
```

Comment:
What happened to the tests?

Reply:
See libdazzle for an example how to run tests under dbus.

Reply:
Actually, adding
```basj
    for f in tests/functional-tests/*.py; do
      patchShebangs "$f"
    done
```

to `postPatch` might be enough. We need to force python2, though.

Reply:
Opened an issue about it, we can probably ignore the tests for now https://bugzilla.gnome.org/show_bug.cgi?id=795594